crates/matrix-sdk-base/src/store/ambiguity_map.rs

```diff
@@ -154,60 +154,76 @@ impl AmbiguityCache {
     }
 
-    async fn get(
-        &mut self,
+    async fn get_old_display_name(
+        &self,
         changes: &StateChanges,
         room_id: &RoomId,
-        member_event: &SyncRoomMemberEvent,
-    ) -> Result<(Option<DisplayNameUsers>, Option<DisplayNameUsers>)> {
+        new_event: &SyncRoomMemberEvent,
+    ) -> Result<Option<String>> {
         use MembershipState::*;
 
         let old_event = if let Some(m) = changes.state.get(room_id).and_then(|events| {
-            events.get(&StateEventType::RoomMember)?.get(member_event.state_key().as_str())
+            events.get(&StateEventType::RoomMember)?.get(new_event.state_key().as_str())
         }) {
             Some(RawMemberEvent::Sync(m.clone().cast()))
         } else {
-            self.store.get_member_event(room_id, member_event.state_key()).await?
+            self.store.get_member_event(room_id, new_event.state_key()).await?
         };
 
-        // FIXME: Use let chains once stable
-        let old_display_name = if let Some(Ok(event)) = old_event.map(|r| r.deserialize()) {
-            if matches!(event.membership(), Join | Invite) {
-                let display_name = if let Some(d) = changes.profiles.get(room_id).and_then(|p| {
-                    p.get(member_event.state_key())?.as_original()?.content.displayname.as_deref()
-                }) {
-                    Some(d.to_owned())
-                } else if let Some(d) = self
-                    .store
-                    .get_profile(room_id, member_event.state_key())
-                    .await?
-                    .and_then(|p| p.into_original()?.content.displayname)
-                {
-                    Some(d)
-                } else {
-                    event.original_content().and_then(|c| c.displayname.clone())
-                };
-
-                Some(display_name.unwrap_or_else(|| event.user_id().localpart().to_owned()))
+        let Some(Ok(old_event)) = old_event.map(|r| r.deserialize()) else { return Ok(None) };
+
+        if matches!(old_event.membership(), Join | Invite) {
+            let display_name = if let Some(d) = changes.profiles.get(room_id).and_then(|p| {
+                p.get(new_event.state_key())?.as_original()?.content.displayname.as_deref()
+            }) {
+                Some(d.to_owned())
+            } else if let Some(d) = self
+                .store
+                .get_profile(room_id, new_event.state_key())
+                .await?
+                .and_then(|p| p.into_original()?.content.displayname)
+            {
+                Some(d)
             } else {
-                None
-            }
+                old_event.original_content().and_then(|c| c.displayname.clone())
+            };
+
+            Ok(Some(display_name.unwrap_or_else(|| old_event.user_id().localpart().to_owned())))
         } else {
-            None
-        };
+            Ok(None)
+        }
+    }
+
+    async fn get_users_with_display_name(
```

Comment:
Can you add a brief doc comment explaining what it does please?

Reply:
f08e545c9923a6c84a1f3376a49bace9ab8704b3